flepimop/gempyor_pkg/src/gempyor/shared_cli.py

```diff
@@ -0,0 +1,213 @@
+        ["-s", "--seir_modifiers_scenarios"],
+        envvar="FLEPI_SEIR_SCENARIO",
+        type=str,
+        default=[],
+        multiple=True,
+        help="override/select the transmission scenario(s) to run",
+    ),
+    "outcome_modifiers_scenarios": click.Option(
+        ["-d", "--outcome_modifiers_scenarios"],
+        envvar="FLEPI_OUTCOME_SCENARIO",
+        type=str,
+        default=[],
+        multiple=True,
+        help="override/select the outcome scenario(s) to run",
+    ),
+    "jobs": click.Option(
+        ["-j", "--jobs"],
+        envvar="FLEPI_NJOBS",
+        type=click.IntRange(min=1),
+        default=multiprocessing.cpu_count(),
+        show_default=True,
+        help="the parallelization factor",
+    ),
+    "nslots": click.Option(
+        ["-n", "--nslots"],
+        envvar="FLEPI_NUM_SLOTS",
+        type=click.IntRange(min=1),
+        help="override the # of simulation runs in the config file",
+    ),
+    "in_run_id": click.Option(
+        ["--in-id", "in_run_id"],
+        envvar="FLEPI_RUN_INDEX",
+        type=str,
+        show_default=True,
+        help="Unique identifier for the run",
+    ),
+    "out_run_id": click.Option(
+        ["--out-id", "out_run_id"],
+        envvar="FLEPI_RUN_INDEX",
+        type=str,
+        show_default=True,
+        help="Unique identifier for the run",
+    ),
+    "in_prefix": click.Option(
+        ["--in-prefix"],
+        envvar="FLEPI_PREFIX",
+        type=str,
+        default=None,
+        show_default=True,
+        help="unique identifier for the run",
+    ),
+    "first_sim_index": click.Option(
+        ["-i", "--first_sim_index"],
+        envvar="FIRST_SIM_INDEX",
+        type=click.IntRange(min=1),
+        default=1,
+        show_default=True,
+        help="The index of the first simulation",
+    ),
+    "stoch_traj_flag": click.Option(
+        ["--stochastic/--non-stochastic", "stoch_traj_flag"],
+        envvar="FLEPI_STOCHASTIC_RUN",
+        default=False,
+        help="Run stochastic simulations?",
+    ),
+    "write_csv": click.Option(
+        ["--write-csv/--no-write-csv"],
+        default=False,
+        show_default=True,
+        help="write csv output?",
+    ),
+    "write_parquet": click.Option(
+        ["--write-parquet/--no-write-parquet"],
+        default=True,
+        show_default=True,
+        help="write parquet output?",
+    ),
+}
+
+
+# adapted from https://stackoverflow.com/a/78533451
+def click_helpstring(params: Union[click.Parameter, List[click.Parameter]]):
+    """
+    A decorator that dynamically appends `click.Parameter`s to the docstring of the decorated function.
+
+    Args:
+        params (Union[click.Parameter, List[click.Parameter]]): A parameter or a list of parameters whose corresponding argument values and help strings will be appended to the docstring of the decorated function.
+
+    Returns:
+        Callable: The original function with an updated docstring.
+    """
```

Comment:
```suggestion
def click_helpstring(
    params: click.Parameter | list[click.Parameter],
) -> Callable[[Callable[..., Any]], Callable[..., Any]]:
    """
    Dynamically append `click.Parameter`s to the docstring of a decorated function.

    Args:
        params: A parameter or a list of parameters whose corresponding argument
            values and help strings will be appended to the docstring of the
            decorated function.

    Returns:
        The original function with an updated docstring.

    Notes:
        Adapted from https://stackoverflow.com/a/78533451.

    Examples:
        >>> import click
        >>> opt = click.Option("-n", "--number", type=int, help="Your favorite number.")
        >>> @click_helpstring(opt)
        ... def test_cli(num: int) -> None:
        ...     print(f"Your favorite number is {num}!")
        ...
        >>> print(test_cli.__doc__)


        Command Line Interface arguments:

        n: Int
    """
```

Reply:
nice.

Reply:
though: can't exclude the redirect to SO - licensing issue.